crates/contract/src/v3_0_2_state.rs

```diff
@@ -40,6 +42,73 @@ impl From<Config> for crate::config::Config {
     }
 }
 
+/// Old version of [`UpdateEntry`] that included votes.
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize)]
+struct UpdateEntry {
+    update: Update,
+    votes: HashSet<AccountId>,
+    bytes_used: u128,
+}
+
+/// Old version of [`ProposedUpdates`] that used the old [`UpdateEntry`] with votes.
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub struct ProposedUpdates {
+    vote_by_participant: IterableMap<AccountId, UpdateId>,
+    entries: IterableMap<UpdateId, UpdateEntry>,
+    id: UpdateId,
+}
+
+impl From<ProposedUpdates> for crate::update::ProposedUpdates {
```

Comment:
Based on https://github.com/near/mpc/pull/1636#discussion_r2615587621 I think this code should not be considered dead code, but the fact that this won't be invoked today is due to a bug where we clear too many update entries when upgrading the contract.

Therefore, I'd like to see this function covered by unit tests to ensure:

1. We don't leave garbage from old iterable maps after the migration.
2. We retain the same set of votes after the migration.